src/main/java/frc/robot/commands/EngageSolenoidCommand.java

```diff
@@ -0,0 +1,57 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj.command.Command;
+import frc.robot.Robot;
+import edu.wpi.first.wpilibj.Timer;
+import frc.robot.controls.DriveController;
+
+public class EngageSolenoidCommand extends Command {
+  public EngageSolenoidCommand() {
+    requires(Robot.liftSystem);
+    // Use requires() here to declare subsystem dependencies
+    // eg. requires(chassis);
+  }
+
+
+  // Called just before this Command runs the first time
+  @Override
+  protected void initialize() {
+   
+  }
+
+  // Called repeatedly when this Command is scheduled to run
+  @Override
+  protected void execute() {
```

Comment:
Will decided not to do that because it takes time to retract the climbing mechanism, so he wanted EngageSolenoidCommand finish() method to return true only after we think the solenoid should be done moving, based on a timer. The same is true of LiftRetractFront. The goal is to make it sequential so that the next command is blocked until the cylinders are done moving. Is that okay?

Reply:
Yeah, just put nothing in execute and use a timer for isFinished(). But you still shouldn't check `if(controller.getStartLift())`